content/en/docs/instrumentation/java/extensibility.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Extensibility
+description: Extensions add new features and capabilities to the agent without having to create a separate distribution or to fork this repository. For example, you can create custom samplers or span exporters, set new defaults, and embed it all in the agent to obtain a single jar file.
+---
+
+## Introduction
+
+Extensions add new features and capabilities to the agent without having to create a separate distribution (for examples and ideas, see [Use cases for extensions](#sample-use-cases)).
+
+The contents in this folder demonstrate how to create an extension for the OpenTelemetry Java instrumentation agent, with examples for every extension point. 
+
+> Read both the source code and the Gradle build script, as they contain documentation that explains the purpose of all the major components.
+
+## Build and add extensions
+
+To build this extension project, run `./gradlew build`. You can find the resulting jar file in `build/libs/`. 
+
+To add the extension to the instrumentation agent:
+
+1. Copy the jar file to a host that is running an application to which you've attached the OpenTelemetry Java instrumentation.
+2. Modify the startup command to add the full path to the extension file. For example:
+
+     ```bash
+     java -javaagent:path/to/opentelemetry-javaagent.jar \
+          -Dotel.javaagent.extensions=build/libs/opentelemetry-java-instrumentation-extension-demo-1.0-all.jar
+          -jar myapp.jar
+     ```
+## Embed extensions in the OpenTelemetry Agent
+
+To simplify deployment, you can embed extensions into the OpenTelemetry Java Agent to produce a single jar file. With an integrated extension, you no longer need the `-Dotel.javaagent.extensions` command line option.
+
+For more information, see the `extendedAgent` task in [build.gradle](build.gradle).
+## Extensions examples
+
+* Custom `IdGenerator`: [DemoIdGenerator](src/main/java/com/example/javaagent/DemoIdGenerator.java)
+* Custom `TextMapPropagator`: [DemoPropagator](src/main/java/com/example/javaagent/DemoPropagator.java)
+* New default configuration: [DemoPropertySource](src/main/java/com/example/javaagent/DemoPropertySource.java)
+* Custom `Sampler`: [DemoSampler](src/main/java/com/example/javaagent/DemoSampler.java)
+* Custom `SpanProcessor`: [DemoSpanProcessor](src/main/java/com/example/javaagent/DemoSpanProcessor.java)
+* Custom `SpanExporter`: [DemoSpanExporter](src/main/java/com/example/javaagent/DemoSpanExporter.java)
+* Additional instrumentation: [DemoServlet3InstrumentationModule](src/main/java/com/example/javaagent/instrumentation/DemoServlet3InstrumentationModule.java)
```

Comment:
All of the local links here are invalid -- i.e., those files don't exist here. Do these files exist somewhere else? If so, link to their origin.